tip-router-operator-cli/src/merkle_root_generator_workflow.rs

```diff
@@ -24,31 +24,82 @@ pub enum MerkleRootGeneratorError {
     let merkle_tree_coll = GeneratedMerkleTreeCollection::new_from_stake_meta_collection(
         stake_meta_coll,
-        Some(rpc_client),
+        None,
     )?;
 
-    write_to_json_file(&merkle_tree_coll, out_path)?;
-    Ok(())
+    Ok(merkle_tree_coll)
 }
 
-fn write_to_json_file(
-    merkle_tree_coll: &GeneratedMerkleTreeCollection,
-    file_path: &PathBuf,
-) -> Result<(), MerkleRootGeneratorError> {
-    let file = File::create(file_path)?;
-    let mut writer = BufWriter::new(file);
-    let json = serde_json::to_string_pretty(&merkle_tree_coll).unwrap();
-    writer.write_all(json.as_bytes())?;
-    writer.flush()?;
-
-    Ok(())
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use solana_sdk::{
+        pubkey::Pubkey,
+        signature::{Keypair, Signer},
+    };
+    use solana_program::hash::Hash;
+    use crate::{StakeMeta, TipDistributionMeta, Delegation};
+
+    #[test]
+    fn test_generate_merkle_root() -> Result<(), MerkleRootGeneratorError> {
+        // Create test stake meta with two delegations
+        let stake_meta = StakeMeta {
+            validator_vote_account: Pubkey::new_unique(),
+            validator_node_pubkey: Pubkey::new_unique(),
+            maybe_tip_distribution_meta: Some(TipDistributionMeta {
+                total_tips: 1_000_000,
+                merkle_root_upload_authority: Keypair::new().pubkey(),
+                tip_distribution_pubkey: Pubkey::new_unique(),
+                validator_fee_bps: 1000,
+            }),
+            delegations: vec![
+                Delegation {
+                    stake_account_pubkey: Pubkey::new_unique(),
+                    staker_pubkey: Pubkey::new_unique(),
+                    withdrawer_pubkey: Pubkey::new_unique(),
+                    lamports_delegated: 1_000_000,
+                },
+                Delegation {
+                    stake_account_pubkey: Pubkey::new_unique(),
+                    staker_pubkey: Pubkey::new_unique(),
+                    withdrawer_pubkey: Pubkey::new_unique(),
+                    lamports_delegated: 2_000_000,
+                },
+            ],
+            total_delegated: 3_000_000,
+            commission: 10,
+        };
+
+        let stake_meta_collection = StakeMetaCollection {
+            epoch: 0,
+            stake_metas: vec![stake_meta],
+            bank_hash: "test_bank_hash".to_string(),
+            slot: 0,
+            tip_distribution_program_id: Pubkey::new_unique(),
+        };
+
+        // Generate merkle tree
+        let merkle_tree_coll = generate_merkle_root(stake_meta_collection)?;
+
+        // Basic validations
+        assert_eq!(merkle_tree_coll.epoch, 0);
+        assert_eq!(merkle_tree_coll.generated_merkle_trees.len(), 1);
+        
+        // Validate generated merkle tree
+        let generated_tree = &merkle_tree_coll.generated_merkle_trees[0];
+        assert!(generated_tree.merkle_root != Hash::default(), "Merkle root should not be zero");
```

Comment:
use `assert_ne!`

Reply:
I would also add an assertion for the actual merkle root given the tree. You could either hard code the hash in once you have things running as expected or calculate the expected value using the same function that does it internally.